### ISSUE TITLE
doesn't work if Kodi userdata path contains non-ASCII chars

### ISSUE THREAD

**Issue by White-Tiger** [bromix/plugin.video.7tv#19]
In my case, my user name includes the char "é" which makes this addon unusable.
I currently have to use Kodi in portable mode to get this and some other addons working (MySpass/Video or something)

```log

13:31:11 T:3392  NOTICE: -->Python Interpreter Initialized<--
13:31:11 T:3392  NOTICE: [plugin.video.7tv] Running: 7TV (2.2.1) on Isengard (15.2) with Python 2.7.8
13:31:11 T:3392   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeDecodeError'>
                                            Error Contents: 'ascii' codec can't decode byte 0xc3 in position 10: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\default.py", line 5, in <module>
                                                runner.run(__provider__)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\runner.py", line 32, in run
                                                __RUNNER__.run(provider, context)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\impl\xbmc\xbmc_runner.py", line 23, in run
                                                results = provider.navigate(context)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\abstract_provider.py", line 123, in navigate
                                                result = method(context, re_match)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\register_provider_path.py", line 12, in wrapper
                                                return func(*args, **kwargs)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\prosiebensat1\provider.py", line 355, in _get_channel_content
                                                channel_image = context.create_resource_path('media', 'channels', '%s.png' % channel_id)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\impl\abstract_context.py", line 153, in create_resource_path
                                                path = os.path.join(self.get_native_path(), 'resources', *path_comps)
                                              File "E:\_\Multimedia\Kodi\system\python\Lib\ntpath.py", line 84, in join
                                                result_path = result_path + p_path
                                            UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 10: ordinal not in range(128)
                                            -->End of Python script error report<--
13:31:11 T:5960   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.7tv/pro7/
13:31:11 T:5960   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.7tv/pro7/) failed
13:31:11 T:1880  NOTICE: Thread LanguageInvoker start, auto delete: false
13:31:11 T:1880  NOTICE: -->Python Interpreter Initialized<--
13:31:11 T:1880  NOTICE: [plugin.video.7tv] Running: 7TV (2.2.1) on Isengard (15.2) with Python 2.7.8
13:31:11 T:4160  NOTICE: Thread BackgroundLoader start, auto delete: false
13:31:14 T:4892  NOTICE: Thread LanguageInvoker start, auto delete: false
13:31:14 T:4892  NOTICE: -->Python Interpreter Initialized<--
13:31:14 T:4892  NOTICE: [plugin.video.7tv] Running: 7TV (2.2.1) on Isengard (15.2) with Python 2.7.8
13:31:14 T:4892   ERROR: EXCEPTION Thrown (PythonToCppException) : -->Python callback/script returned the following error<--
                                             - NOTE: IGNORING THIS CAN LEAD TO MEMORY LEAKS!
                                            Error Type: <type 'exceptions.UnicodeDecodeError'>
                                            Error Contents: 'ascii' codec can't decode byte 0xc3 in position 10: ordinal not in range(128)
                                            Traceback (most recent call last):
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\default.py", line 5, in <module>
                                                runner.run(__provider__)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\runner.py", line 32, in run
                                                __RUNNER__.run(provider, context)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\impl\xbmc\xbmc_runner.py", line 23, in run
                                                results = provider.navigate(context)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\abstract_provider.py", line 123, in navigate
                                                result = method(context, re_match)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\register_provider_path.py", line 12, in wrapper
                                                return func(*args, **kwargs)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\prosiebensat1\provider.py", line 355, in _get_channel_content
                                                channel_image = context.create_resource_path('media', 'channels', '%s.png' % channel_id)
                                              File "C:\Users\RenꝁppData\Roaming\Kodi\addons\plugin.video.7tv\resources\lib\kodion\impl\abstract_context.py", line 153, in create_resource_path
                                                path = os.path.join(self.get_native_path(), 'resources', *path_comps)
                                              File "E:\_\Multimedia\Kodi\system\python\Lib\ntpath.py", line 84, in join
                                                result_path = result_path + p_path
                                            UnicodeDecodeError: 'ascii' codec can't decode byte 0xc3 in position 10: ordinal not in range(128)
                                            -->End of Python script error report<--
13:31:14 T:5960   ERROR: XFILE::CDirectory::GetDirectory - Error getting plugin://plugin.video.7tv/pro7/
13:31:14 T:5960   ERROR: CGUIMediaWindow::GetDirectory(plugin://plugin.video.7tv/pro7/) failed
13:31:14 T:4776  NOTICE: Thread LanguageInvoker start, auto delete: false
13:31:14 T:4776  NOTICE: -->Python Interpreter Initialized<--
```

**Comment by bromix** [comment]
Please test the 3.0.0alpha from my repo and confirm if the problem still exists.

**Comment by White-Tiger** [comment]
I could probably test it if you could provide me a link to said version (optionally with a short explanation on how to add it to Kodi)

**Comment by bromix** [comment]
Can you please test version 3.0.0

**Comment by White-Tiger** [comment]
well... you still didn't mention how to install your plugin... yet I've figured it out as it is really just the zip version of the "bare" repository... (you should probably put a note into a README.md)

The original issue seems to be gone, but I've failed to playback videos, at least some times... I'll get the new 7TV `opening..`, after that a `Working...` from Kodi and finally `one or more items failed to play` (during all of this, Kodi isn't very responsible)
```
22:28:08 T:31392  NOTICE: -->Python Interpreter Initialized<--
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.search.size" to "addon.search.size
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.view.override" to "view.override
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.setup_wizard" to "addon.setup
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.view.episodes" to "view.override.episodes
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.content.max_per_page" to "addon.items.per_page
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.fanart.show" to "addon.fanart.show
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.cache.size" to "addon.cache.size
22:28:08 T:31392  NOTICE: [plugin.video.7tv] Converting setting "kodion.view.default" to "view.override.default
22:28:08 T:31396  NOTICE: Thread BackgroundLoader start, auto delete: false
22:28:10 T:31400  NOTICE: Thread LanguageInvoker start, auto delete: false
22:28:10 T:31400  NOTICE: -->Python Interpreter Initialized<--
22:28:10 T:31400  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:28:10 T:31404  NOTICE: Thread BackgroundLoader start, auto delete: false
22:28:13 T:31428  NOTICE: Thread LanguageInvoker start, auto delete: false
22:28:13 T:31428  NOTICE: -->Python Interpreter Initialized<--
22:28:13 T:31428  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:28:13 T:31432  NOTICE: Thread BackgroundLoader start, auto delete: false
22:28:14 T:31440  NOTICE: Thread FileCache start, auto delete: false
22:28:16 T:31456  NOTICE: Previous line repeats 2 times.
22:28:16 T:31456  NOTICE: Thread LanguageInvoker start, auto delete: false
22:28:16 T:31456  NOTICE: -->Python Interpreter Initialized<--
22:28:16 T:31456  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:28:18 T:28612   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.7tv/play/?video_id=3625493]
22:28:18 T:31472  NOTICE: Thread BackgroundLoader start, auto delete: false
22:28:20 T:31476  NOTICE: Thread LanguageInvoker start, auto delete: false
22:28:20 T:31476  NOTICE: -->Python Interpreter Initialized<--
22:28:20 T:31476  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:28:33 T:28612  NOTICE: DVDPlayer: Opening: http://vod-level3-psd-progressive.p7s1digital.de/clips/01/e1/4029569-4oz4s4-tp12.mp4?token=00~3d7c81~3d7c81~5601b9e3~15180~0-a136b35425982d548f2ac541f&ts=1442953699
22:28:33 T:28612 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
22:28:33 T:31508  NOTICE: Thread DVDPlayer start, auto delete: false
22:28:33 T:31508  NOTICE: Creating InputStream
22:28:34 T:31512  NOTICE: Thread FileCache start, auto delete: false
22:28:34 T:31508  NOTICE: Creating Demuxer
22:29:04 T:31508   ERROR: CDVDDemuxFFmpeg::Open - Error, could not open file http://vod-level3-psd-progressive.p7s1digital.de/clips/01/e1/4029569-4oz4s4-tp12.mp4?token=00~3d7c81~3d7c81~5601b9e3~15180~0-a136b35425982d548f2ac541f&ts=1442953699
22:29:04 T:31508   ERROR: CDVDPlayer::OpenDemuxStream - Error creating demuxer
22:29:04 T:31508  NOTICE: CDVDPlayer::OnExit()
22:29:04 T:28612   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.7tv/play/?video_id=4029569]
22:29:04 T:28612  NOTICE: CDVDPlayer::CloseFile()
22:29:04 T:28612  NOTICE: DVDPlayer: waiting for threads to exit
22:29:04 T:28612  NOTICE: DVDPlayer: finished waiting
22:29:04 T:28612  NOTICE: CDVDPlayer::CloseFile()
22:29:04 T:28612  NOTICE: DVDPlayer: waiting for threads to exit
22:29:04 T:28612  NOTICE: DVDPlayer: finished waiting
22:29:04 T:31568  NOTICE: Thread JobWorker start, auto delete: true
22:29:06 T:31580  NOTICE: Previous line repeats 1 times.
22:29:06 T:31580  NOTICE: Thread BackgroundLoader start, auto delete: false
22:29:06 T:31584  NOTICE: Thread JobWorker start, auto delete: true
22:29:06 T:31592  NOTICE: Previous line repeats 1 times.
22:29:06 T:31592  NOTICE: Thread FileCache start, auto delete: false
22:29:08 T:31604  NOTICE: Previous line repeats 2 times.
22:29:08 T:31604  NOTICE: Thread LanguageInvoker start, auto delete: false
22:29:08 T:31604  NOTICE: -->Python Interpreter Initialized<--
22:29:08 T:31604  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:29:10 T:28612   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.7tv/play/?video_id=3625124]
22:29:10 T:31608  NOTICE: Thread BackgroundLoader start, auto delete: false
22:29:11 T:31612  NOTICE: Thread FileCache start, auto delete: false
22:29:12 T:31620  NOTICE: Previous line repeats 1 times.
22:29:12 T:31620  NOTICE: Thread LanguageInvoker start, auto delete: false
22:29:12 T:31620  NOTICE: -->Python Interpreter Initialized<--
22:29:12 T:31620  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:29:22 T:28612  NOTICE: DVDPlayer: Opening: http://vod-level3-psd-progressive.p7s1digital.de/clips/04/e4/3907182-1odvxok-tp12.mp4?token=00~3b9e6e~3b9e6e~5601ba17~15180~0-4200808cfadd51d8f72925a36&ts=1442953751
22:29:22 T:28612 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
22:29:22 T:31636  NOTICE: Thread DVDPlayer start, auto delete: false
22:29:22 T:31636  NOTICE: Creating InputStream
22:29:22 T:31636  NOTICE: Creating Demuxer
22:29:22 T:31640  NOTICE: Thread FileCache start, auto delete: false
22:29:52 T:31636   ERROR: CDVDDemuxFFmpeg::Open - Error, could not open file http://vod-level3-psd-progressive.p7s1digital.de/clips/04/e4/3907182-1odvxok-tp12.mp4?token=00~3b9e6e~3b9e6e~5601ba17~15180~0-4200808cfadd51d8f72925a36&ts=1442953751
22:29:52 T:31636   ERROR: CDVDPlayer::OpenDemuxStream - Error creating demuxer
22:29:52 T:31636  NOTICE: CDVDPlayer::OnExit()
22:29:52 T:28612   ERROR: Playlist Player: skipping unplayable item: 0, path [plugin://plugin.video.7tv/play/?video_id=3907182]
22:29:52 T:28612  NOTICE: CDVDPlayer::CloseFile()
22:29:52 T:28612  NOTICE: DVDPlayer: waiting for threads to exit
22:29:52 T:28612  NOTICE: DVDPlayer: finished waiting
22:29:52 T:28612  NOTICE: CDVDPlayer::CloseFile()
22:29:52 T:28612  NOTICE: DVDPlayer: waiting for threads to exit
22:29:52 T:28612  NOTICE: DVDPlayer: finished waiting
22:29:52 T:31736  NOTICE: Thread JobWorker start, auto delete: true
22:34:36 T:28236  NOTICE: Previous line repeats 1 times.
22:34:36 T:28236  NOTICE: Thread BackgroundLoader start, auto delete: false
22:34:36 T:29748  NOTICE: Thread JobWorker start, auto delete: true
22:34:38 T:29408  NOTICE: Thread LanguageInvoker start, auto delete: false
22:34:38 T:29408  NOTICE: -->Python Interpreter Initialized<--
22:34:39 T:29408  NOTICE: [plugin.video.7tv] Running: 7TV (3.0.0) on Isengard (15.2) with Python (2.7.8)
22:34:57 T:28612  NOTICE: DVDPlayer: Opening: http://vod-level3-psd-progressive.p7s1digital.de/clips/0a/4a/3881139-5ka0lw-tp12.mp4?token=00~3b38b3~3b38b3~5601bb61~15180~0-c57d7be1e3dfd54f0ab1c20a1&ts=1442954081
22:34:57 T:28612 WARNING: CDVDMessageQueue(player)::Put MSGQ_NOT_INITIALIZED
22:34:57 T:30680  NOTICE: Thread DVDPlayer start, auto delete: false
22:34:57 T:30680  NOTICE: Creating InputStream
22:34:57 T:30680  NOTICE: Creating Demuxer
22:34:57 T:30780  NOTICE: Thread FileCache start, auto delete: false
22:35:15 T:30680  NOTICE: Opening stream: 0 source: 256
22:35:15 T:30680  NOTICE: Creating video codec with codec id: 28
22:35:15 T:30680  NOTICE: CDVDVideoCodecFFmpeg::Open() Using codec: H.264 / AVC / MPEG-4 AVC / MPEG-4 part 10
22:35:15 T:30680  NOTICE: Creating video thread
22:35:15 T:30680  NOTICE: Opening stream: 1 source: 256
22:35:15 T:30680  NOTICE: Finding audio codec for: 86018
22:35:15 T:31308  NOTICE: Thread DVDPlayerVideo start, auto delete: false
22:35:15 T:31308  NOTICE: running thread: video_thread
22:35:15 T:30680  NOTICE: Creating audio thread
22:35:15 T:31336  NOTICE: Thread DVDPlayerAudio start, auto delete: false
22:35:15 T:31336  NOTICE: running thread: CDVDPlayerAudio::Process()
22:35:15 T:31336  NOTICE: Creating audio stream (codec id: 86018, channels: 2, sample rate: 48000, no pass-through)
22:35:15 T:31336  NOTICE: CDVDPlayerAudio::OutputPacket duplicate 4 packets of duration 21
22:35:15 T:31308  NOTICE: CDVDVideoCodecFFmpeg::GetFormat - Creating DXVA(1280x720)
22:35:16 T:31308  NOTICE:  fps: 25.000000, pwidth: 1280, pheight: 720, dwidth: 1280, dheight: 720
22:35:16 T:31308  NOTICE: Display resolution DESKTOP : 1920x1080@ 60.00 - Full Screen (16)
22:35:16 T:31308  NOTICE: D3D: rendering method forced to DXVA processor
22:35:16 T:28612 WARNING: CWinRenderer::UpdateVideoFilter - chosen scaling method 1 is not supported by renderer```

Basically, the old version did the job a lot faster...

**Comment by bromix** [comment]
Arg...the update comes with the official KODI repo.

Please remove all of 7TV (also the user data)

Please provide ALWAYS complete logs via pastebin.com or similar.

I've to see your platform which seems not to support the format for HD content.

**Comment by White-Tiger** [comment]
when will the update be officially available? There wasn't any the time I tried.
I'm on Win 8.1 x86_64 with an NVIDIA GPU

**Comment by bromix** [comment]
Since yesterday. Trigger the Update via the sidebar in KODI.

**Comment by White-Tiger** [comment]
sry for this late reply, everything seems to be working now, with Kodi 15.2 RC2
So it's working as fast as the old 7TV IIRC

**Comment by bromix** [comment]
Cool...good to know. So we can close this issue?

**Comment by White-Tiger** [comment]
Yep and oh.. didn't see that even I can close it :P sry about that.
Thanks a lot for fixing though ;)
~closed~
